index.js

```diff
@@ -63,7 +63,7 @@ const fastifyReplyFrom = fp(function from (fastify, opts, next) {
     }
 
     // we leverage caching to avoid parsing the destination URL
-    const dest = getUpstream(req, base)
+    const dest = getUpstream(this.request, base)
```

Comment:
We need a test for this change.

Reply:
There's already a test checking the type, is there something else you wanted to test for?

Reply:
If the current tests do not show a problem then they do not cover the change.

Reply:
I had to edit the tests to check for the new type, see my edits to `types/index.test-d.ts`

Reply:
This is a change to the actual code. There needs to be an associated unit test to guard against regression and assure that the data is being passed correctly.

Reply:
Alright, how do you want me to check whether the argument is a fastify request?

Reply:
That's the core of the work. How did you determine that it needs to be passed in? Write a test that fails your expectation without the change.

Reply:
right now: `req` is equals to `this.request.raw`, `this.request` is the fastify request.

In any case:

- this PR would be a breaking change
- we need to update docs
- tests must be updated to check that option functions get a fastify object

Reply:
I've added tests that check that the request passed in the callback is correct, and verified that those tests fail on master.

In terms of docs, I read through the README, but the docs never specified exactly what the type of the request parameter is, so I left it as-is. Do you want me to add that?

Reply:
Please add it, thanks!

Reply:
done